### ISSUE TITLE
False Positive: [EA] Unnecessary assignment, a nested call would simplify workflow.

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1450]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) 3.0.5 |
| Language level | PHP 7.3 |

Current behaviour
--
When using a reference for easy access to a value deep in an array.

Before:
![Screenshot from 2019-11-20 16-44-57](https://user-images.githubusercontent.com/866668/69253546-1b1b9500-0bb5-11ea-9de6-1c9377e2dd35.png)

```
  $a = ['b' => ['c' => ['d' => "123.456"]]];
  $ref1 = &$a['b']['c']['d'];
  $ref1 = trim($ref1);
```
generates the error/inspection: [EA] Unnecessary assignment, a nested call would simplify workflow.
After applying sugested fix:
```
  $a = ['b' => ['c' => ['d' => "123.456"]]];
  $ref1 = trim($a['b']['c']['d']);
```

Expected behaviour
--
No warning for referenced variables

Environment details
---
PhpStorm 2019.3 Beta
Build #PS-193.5233.21, built on November 14, 2019
PhpStorm EAP User
Expiration date: December 14, 2019
Runtime version: 11.0.4+10-b520.6 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.3.0-23-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1923M
Cores: 8
Registry: run.processes.with.pty=TRUE, documentation.show.toolbar=true
Non-Bundled Plugins: Shell Process, String Manipulation, SvgViewer 2, com.aurimasniekis.phppsr4namespacedetector, com.kalessil.phpStorm.phpInspectionsUltimate, com.raket.silverstripe, com.wix.eslint, de.espend.idea.laravel, in.xiv.cs, izhangzhihao.rainbow.brackets, name.kropp.intellij.makefile, net.king2500.plugins.PhpAdvancedAutoComplete, net.seesharpsoft.intellij.plugins.csv, net.vektah.codeglance, org.intellij.RegexpTester, org.psliwa.idea.composer

**Comment by kalessil** [comment]
Fixed!
